lib/pure/asyncmacro.nim

```diff
@@ -332,8 +253,24 @@ proc asyncSingleProc(prc: NimNode): NimNode {.compileTime.} =
     if returnType.kind == nnkEmpty:
       # Add Future[void]
       result.params[0] = parseExpr("owned(Future[void])")
+
+  # based on the yglukhov's patch to chronos: https://github.com/status-im/nim-chronos/pull/47
```

Comment:
I would remove this comment, at this point this really doesn't have many parallels. I doubt it'll be useful for those exploring this code.